### PR TITLE
Restrict bookings to future dates only

### PR DESCRIPTION
**Pull request by luciademoja**
Title of the PT story: Restrict bookings to future dates only

Link to PT story: https://www.pivotaltracker.com/story/show/129743513

Description of the PR:

1. Validates the date on server throwing an error when booking passed dates
2. Changes color of time slots to grey when the date is in the past
3. Shows modal error message when clicking on a grey slot

skarmavbild 2016-09-12 kl 08 48 50

skarmavbild 2016-09-12 kl 20 03 21

Still missing the front end modal display message.

Ready for review!
@tochman @diraulo @MikaelFeher @lollypop27

### CODE REVIEW

File: app/assets/javascripts/dashboard.js [hunk 1]
```diff
@@ -61,14 +61,22 @@ function updateCurrentBookings(response) {
...
     });
-
 }
 
 function getBackgroundColor(obj) {
+  var date = getDispalyedDate();
+  var newDate = new Date(date + " 00:00");
     var color = (obj.state == 'booked') ? 'red' : 'green';
+    if (newDate < today()) {
+    color = 'grey'
```
Comment (diraulo) [inline app/assets/javascripts/dashboard.js]: indent code here and add a semicolon.

File: app/assets/javascripts/dashboard.js [hunk 2]
```diff
@@ -61,14 +61,22 @@ function updateCurrentBookings(response) {
             $(card + " .content .with-scroll").append('<div class="action" id="action_' + slot.info.id + '" style="background-color: ' + getBackgroundColor(slot) + '">' + getInfo(slot) + '</div>');
         });
     });
-
 }
 
 function getBackgroundColor(obj) {
+  var date = getDispalyedDate();
```
Comment (diraulo) [inline app/assets/javascripts/dashboard.js]: fix the code formatting on this function

File: app/assets/javascripts/dashboard.js [hunk 3]
```diff
@@ -64,34 +64,27 @@ function updateCurrentBookings(response) {
...
-
 function today() {
-  return new Date(new Date().toISOString().slice(0,10) + " 00:00")
+  return new Date(new Date().toJSON().slice(0,10) + " 00:00");
 }
 
 function getInfo(obj) {
-    var message = (obj.info.client != null) ? setSlotMessage(obj) : obj.info.time;
-    return message;
+  var message = (obj.info.client !== null) ? setSlotMessage(obj) : obj.info.time;
```
Comment (luciademoja) [inline app/assets/javascripts/dashboard.js]: @diraulo that `=`sign you added actually fks up everything and the resources don't display correctly anymore. It goes back to normal if I take it back. Is there a reason why we should keep it?

Comment (diraulo) [inline app/assets/javascripts/dashboard.js]: @luciademoja you can remove it

**Pull request merged**